### ISSUE TITLE
Allow updating connections from control plane

### ISSUE THREAD

**Issue by mouradmourafiq** [polyaxon/polyaxon#1150]
### Use case

Apart from the `artifactsStore` other connections could be updated without an agent upgrade.

### Feature description

Add possibility to update connections from the control plane, in which case the agent should discard the local definition and only consider the control plane values.

Add a flag to the agent API that shows that the user is opting for control plane connections management. 

This change only impacts the connections schema definition, the user still needs to define secrets on the on-prem cluster.

### Alternatives

Local definition should be still possible.

**Comment by denisoliveirac** [comment]
It may be cool to think of a way to test the connection during definition.

**Comment by mouradmourafiq** [comment]
Makes sense, that should be easy for some connections, sending a job for instance to perform `ls` on an S3 bucket, or slack test job.
